client/batch.go

```diff
@@ -308,6 +315,29 @@ func (b *Batch) Scan(s, e interface{}, maxRows int64) {
 	b.initResult(1, 0, nil)
 }
 
+// ReverseScan retrieves the rows in reserve order starting at end (exclusive)
+// and finishing at begin(inclusive).
+//
+// A new result will be appended to the batch which will contain up to maxRows
+// rows and Result.Err will indicate success or failure.
+//
+// key can be either a byte slice, a string, a fmt.Stringer or an
+// encoding.BinaryMarshaler.
+func (b *Batch) ReverseScan(s, e interface{}, maxRows int64) {
```

Comment:
Yes ,I have considered that. In some SQL  engine like sqlite4, the end key of the reverse scan is always inclusive, and expose key.Next() to it is not a good idea. I think it is better to change the semantics of reverse scan to start key exclusive and end key inclusive to client interface. But internally our implementation is opposite by conversion the key and end key before send in kv/db.go. What`s your opinion?